src/server/game/Quests/WorldQuestMgr.cpp

```diff
@@ -0,0 +1,654 @@
+                    AddWorldQuestTemplate(quest, worldQuestTemplate);
+                }
+            }
+        }
+    }
+
+    if (_emissaryWorldQuestTemplates.size() < WORLD_QUEST_EMISSARY)
+        TC_LOG_ERROR("server.loading", "World Quest: There is %lu emissary quests but %u needed...", _emissaryWorldQuestTemplates.size(), uint32(WORLD_QUEST_EMISSARY));
+}
+
+void WorldQuestMgr::AddWorldQuestTemplate(Quest const* quest, WorldQuestTemplate* worldQuestTemplate)
+{
+    uint8 questTeamId = GetQuestTeamId(quest);
+    uint32 questId = quest->GetQuestId();
+    int32 questExpansion = quest->GetExpansion();
+
+    switch (questTeamId)
+    {
+        case TEAM_NEUTRAL:
+        {
+            _worldQuestTemplates[questExpansion][TEAM_ALLIANCE][questId] = worldQuestTemplate;
+            _worldQuestTemplates[questExpansion][TEAM_HORDE][questId] = worldQuestTemplate;
+
+            if (quest->IsEmissaryQuest())
+            {
+                _emissaryWorldQuestTemplates[questExpansion][TEAM_ALLIANCE][questId] = worldQuestTemplate;
+                _emissaryWorldQuestTemplates[questExpansion][TEAM_HORDE][questId] = worldQuestTemplate;
+            }
+
+            break;
+        }
+        default:
+        {
+            _worldQuestTemplates[questExpansion][questTeamId][questId] = worldQuestTemplate;
+
+            if (quest->IsEmissaryQuest())
+                _emissaryWorldQuestTemplates[questExpansion][questTeamId][questId] = worldQuestTemplate;
+
+            break;
+        }
+    }
+}
+
+void WorldQuestMgr::LoadWorldQuestRewardTemplates()
+{
+    QueryResult result = WorldDatabase.Query("SELECT id, questType, rewardType - 1 AS rewardType, rewardId, rewardCount, rewardContext FROM world_quest_reward");
+    if (!result)
+        return;
+
+    do
+    {
+        Field* fields = result->Fetch();
+
+        WorldQuestReward worldQuestReward = WorldQuestReward(fields[0].GetUInt32(),
+                                                             fields[1].GetUInt8(),
+                                                             fields[2].GetUInt8(),
+                                                             fields[3].GetUInt32(),
+                                                             fields[4].GetUInt32(),
+                                                             fields[5].GetUInt32());
+
+        _worldQuestRewards[worldQuestReward.Id].push_back(worldQuestReward);
+        _worldQuestRewardByQuestInfos[worldQuestReward.QuestType].push_back(worldQuestReward.Id);
+    } while (result->NextRow());
+}
+
+void WorldQuestMgr::LoadActiveWorldQuests()
+{
+    // not asynch, only at startup
+    CharacterDatabasePreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_WORLD_QUEST);
+    PreparedQueryResult result = CharacterDatabase.Query(stmt);
+
+    if (!result)
+    {
+        TC_LOG_INFO("server.loading", "World Quest: There is no active world quests.");
```

Comment:
There **are** no active world quests

Reply:
... or "World Quest: No world quest is currently active." (optional phrasing)